src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -137,798 +34,233 @@ namespace JSONHelpers
+            return info;
         }
         catch (const winrt::hresult_error&)
         {
-            return false;
+            return std::nullopt;
         }
     }
 
-    json::JsonArray FancyZonesData::SerializeAppZoneHistory() const
+    json::JsonObject GridLayoutInfoJSON::ToJson(const FancyZonesDataTypes::GridLayoutInfo& gridInfo)
     {
-        std::scoped_lock lock{ dataLock };
-        json::JsonArray appHistoryArray;
+        json::JsonObject infoJson;
+        infoJson.SetNamedValue(L"rows", json::value(gridInfo.m_rows));
+        infoJson.SetNamedValue(L"columns", json::value(gridInfo.m_columns));
+        infoJson.SetNamedValue(L"rows-percentage", NumVecToJsonArray(gridInfo.m_rowsPercents));
+        infoJson.SetNamedValue(L"columns-percentage", NumVecToJsonArray(gridInfo.m_columnsPercents));
 
-        for (const auto& [appPath, appZoneHistoryData] : appZoneHistoryMap)
+        json::JsonArray cellChildMapJson;
+        for (int i = 0; i < gridInfo.m_cellChildMap.size(); ++i)
         {
-            appHistoryArray.Append(AppZoneHistoryJSON::ToJson(AppZoneHistoryJSON{ appPath, appZoneHistoryData }));
+            cellChildMapJson.Append(NumVecToJsonArray(gridInfo.m_cellChildMap[i]));
         }
+        infoJson.SetNamedValue(L"cell-child-map", cellChildMapJson);
 
-        return appHistoryArray;
+        return infoJson;
     }
 
-    bool FancyZonesData::ParseDeviceInfos(const json::JsonObject& fancyZonesDataJSON)
+    std::optional<FancyZonesDataTypes::GridLayoutInfo> GridLayoutInfoJSON::FromJson(const json::JsonObject& infoJson)
     {
-        std::scoped_lock lock{ dataLock };
         try
         {
-            auto devices = fancyZonesDataJSON.GetNamedArray(L"devices");
+            FancyZonesDataTypes::GridLayoutInfo info(FancyZonesDataTypes::GridLayoutInfo::Minimal{});
 
-            for (uint32_t i = 0; i < devices.Size(); ++i)
-            {
-                if (auto device = DeviceInfoJSON::DeviceInfoJSON::FromJson(devices.GetObjectAt(i)); device.has_value())
-                {
-                    deviceInfoMap[device->deviceId] = std::move(device->data);
-                }
-                else
-                {
-                    return false;
-                }
-            }
-
-            return true;
-        }
-        catch (const winrt::hresult_error&)
-        {
-            return false;
-        }
-    }
+            info.m_rows = static_cast<int>(infoJson.GetNamedNumber(L"rows"));
+            info.m_columns = static_cast<int>(infoJson.GetNamedNumber(L"columns"));
 
-    json::JsonArray FancyZonesData::SerializeDeviceInfos() const
-    {
-        std::scoped_lock lock{ dataLock };
-        json::JsonArray DeviceInfosJSON{};
+            json::JsonArray rowsPercentage = infoJson.GetNamedArray(L"rows-percentage");
+            json::JsonArray columnsPercentage = infoJson.GetNamedArray(L"columns-percentage");
+            json::JsonArray cellChildMap = infoJson.GetNamedArray(L"cell-child-map");
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.